macOS/DuckDuckGo/Tab/View/BrowserTabViewController.swift

```diff
@@ -598,26 +598,43 @@ final class BrowserTabViewController: NSViewController {
                 }
                 return old == new
             })
-            .map { [weak tabViewModel] tabContent -> AnyPublisher<Void, Never> in
+            .map { [weak self, tabViewModel] tabContent -> AnyPublisher<Void, Never> in
                 // For non-URL tabs, just emit an event displaying the tab content
                 guard let tabViewModel, tabContent.isUrl else {
                     return Just(()).eraseToAnyPublisher()
                 }
 
-                // For URL tabs, we only want to show tab content (webView) when
-                // it has content to display (first navigation had been committed)
-                // or starts navigation.
-                return Publishers.Merge(
-                    tabViewModel.tab.$hasCommittedContent
-                        .filter { $0 == true }
-                        .asVoid(),
-                    tabViewModel.tab.navigationStatePublisher.compactMap { $0 }
-                        .filter{ $0 >= .started }
-                        .asVoid()
-                )
-                // take the first such event and move forward.
-                .prefix(1)
-                .eraseToAnyPublisher()
+                // If the current content is the native internal site, delay the webview presentation
+                // until a website renders (or edge cases) to avoid white flash
+                // Enabled for internal users for test
```

Comment:
Nitpick about documentation – this line may get outdated quickly and as such, we could remove it.

I think that ultimately we would be interested in rolling out this change behind a Privacy Config flag to be able to disable it remotely if needed. So the flag will stay here long-term, and we’d only change its source in FeatureFlagDescribing protocol.

Reply:
Agree 👍